woocommerce/woocommerce-gutenberg-products-block
# Featured Product: Enable global style

This PR is blocked by #5542

This PR adds support for global style for the `Featured Product block`.

Now, the user can edit the style for:
- text-color
- background-color
- font-size
- border-color
- border-radius
- border-width


## What missing:

### Duotone

At this stage, the `Duotone` feature is implemented with the `filter` CSS property. The problem is that on this block, the image is implemented like `background-image`, so we can't use the `filter` property.

### Padding

At this stage, with the current global style API, the padding is applied to the entire wrapper, so I guess that it doesn't make sense enable it.




#4965 

### Screenshots


![image](https://user-images.githubusercontent.com/4463174/149181700-1385da24-c1cc-42a6-ac4e-1cb98c9a6bb3.png)
*without global style*


![image](https://user-images.githubusercontent.com/4463174/149180767-c0f60ebf-07dc-46f9-95a1-8d90a6f7d7e9.png)
*with global style*


### Testing

### Manual Testing

Check out this branch:

1.  On `WordPress 5.9`, install and enable the `Gutenberg` plugin.
2. Install and enable the `Twenty Twenty-Two` theme.
3. Add the `Featured Product block` to a post.
4. On the right sidebar, personalize the styles of the block.
5. Go on the page and check if there are changes.
6. Reset to default using the `Reset` button from the different sections. 
7. Go to Dashboard and select Appearance > Editor (beta). On top of the screen, select Home > Browser all templates > Single Post. When the page is loaded, add the block to the page.
8. On the Editor page click on the `Styles` icon on the right-top corner.
9. Verify that the `Featured Product block` is shown under the `Blocks` section. Personalize again the block.
10. Save your changes.
11. Go on the page created earlier and check if all styles are applied correctly.
12. Edit your previous post/page again.
13. Change again the styles.
14. Save your changes.
15. Check if these styles have priority over the styles from the Site Editor.

### Changelog

> Add support for the global style for the Featured Product block.
